akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -4,17 +4,39 @@
 
 package akka.http.impl.engine.http2
 
+import scala.annotation.switch
+
 /** INTERNAL API */
 private[akka] object Http2Compliance {
+  def parseSettungsEnablePushValue(incomingValue: Int): Boolean =
+    (incomingValue: @switch) match {
+      case 0 ⇒ false
+      case 1 ⇒ true
+      case _ ⇒ throw new IllegalEnablePushValueSettingException(s"Illegal value in SETTINGS_ENABLE_PUSH settings header! MUST BE 0 or 1, but was: [$incomingValue]")
+    }
+
+  private val MinOutMaxFrameSize = Math.pow(2, 14).toInt // minimum allowed to be set
+  private val MaxOutMaxFrameSize = (Math.pow(2, 24) - 1).toInt // maximum allowed to be set
+
+  final class WindowOverflowException(currentWindow: Int, increment: Int)
```

Comment:
I was somewhat thinking we could expand and collapse them, in the sense of look how exceptions need to be reacted on and then collapse them together into a minimum viable set or maybe some common super types.

For example some errors need to be rendered as PROTOCOL_ERRORS, some as FRAME_SIZE_ERROR and some as others... Perhaps we'll simply carry the error code to GOAWAY with in the exception, that would solve it as well :)

I can give it a shot at collapsing using that idea